src/IDCodeValidator.java

```diff
@@ -0,0 +1,193 @@
+        if (dayNumber > 0 && dayNumber <= 31) {
+            System.out.println(dayNumber);
+            return true;
+        }
+        System.out.println("There is no such day");
+        return false;
+    }
+
+    public static boolean isMonthNumberCorrect(int monthNumber) {
+        if (monthNumber > 0 && monthNumber <= 12) {
+            System.out.println(monthNumber);
+            return true;
+        }
+        System.out.println("There is no such month");
+        return false;
+    }
+
+
+    public static boolean isBirthDateCorrect(int yearNumber, int monthNumber, int dayNumber) {
+        boolean result;
+        int[] numbersOfDays = {0, 31, 28, 31, 30, 31, 30, 31, 30, 31, 30, 31};
+        if (numbersOfDays[2] == 28 && isLeapYear(yearNumber)) {
+            result = (dayNumber <= numbersOfDays[monthNumber] + 1);
+        } else {
+            result = (dayNumber <= numbersOfDays[monthNumber]);
+        }
+        if (result) {
+            System.out.println("Your birth date is correct");
+        } else {
+            System.out.println("Your birth date is not correct");
+        }
+        return result;
+    }
+
+
+    public static boolean isLeapYear(int yearNumber) {
+        if ((yearNumber % 4 == 0 && yearNumber % 100 != 0) || (yearNumber % 400 == 0)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    public static boolean checkControlNumber(String idCode) {
+        int result = 0;
+        List<Integer> a = new ArrayList<>();
+        a.add(Integer.valueOf(idCode));
+        for (int i = 0; i < idCode.length(); i++) {
+            String n = String.valueOf(idCode.charAt(i));
+            a.add(Integer.parseInt(n));
+        }
+        List<Integer> x = new ArrayList<>();
+        x.add(1);
+        x.add(2);
+        x.add(3);
+        x.add(4);
+        x.add(5);
+        x.add(6);
+        x.add(7);
+        x.add(8);
+        x.add(9);
+        x.add(1);
+
+        List<Integer> k = new ArrayList<>();
+        k.add(3);
+        k.add(4);
+        k.add(5);
+        k.add(6);
+        k.add(7);
+        k.add(8);
+        k.add(9);
+        k.add(1);
+        k.add(2);
+        k.add(3);
+
+
+        for (int i = 0; i < a.size() - 1; i++) {
+            result = result + (a.get(i) * x.get(i));
+        }
+        int leftover = result % 11;
+
+
+        if (leftover < 10 && result == idCode.length() - 1) {
+            System.out.println(result);
+        }
+        if (leftover == 10) {
+            for (int i = 0; i < a.size() - 1; i++) {
+                int result1 = result + (a.get(i) * k.get(i));
+                System.out.println(result1);
+
+                if (result1 < 10 && result1 == idCode.length() - 1) {
+                    System.out.println(result1);
+                }
+                if (result1 == 10 && idCode.length() - 1 == 0) {
+                    System.out.println(result1);
+                }
+
+            }
```

Comment:
Don't check for things inside the loop.
It's the same logic as the first loop. We need to get the sum of all numbers from the id code (excluding control number) multiplied by the weight (elements in list called k).
Also, it's a completely new calculation, so don't do
`int result1 = result + (a.get(i) * k.get(i));`
declare new variabe secondResult before the loop or set the value of `result` to 0 before the loop, this way you can reuse it.